fe/fe-core/src/main/java/com/starrocks/server/CatalogMgr.java

```diff
@@ -74,11 +74,11 @@ public synchronized void createCatalog(CreateCatalogStmt stmt) throws DdlExcepti
         writeLock();
```

Comment:
since it's a synchronized method, why still use lock here?

Reply:
Good catch!

Reply:
Synchronized of create method is mainly used to ensure that create cannot be executed in multiple threads at the same time. In addition, I need to ensure that when reading catalogs, It can't be written at the same time, so `catalogExists(String catalogName)` added a read lock.

Reply:
I think the concurrency can be controlled by one lock.

Reply:
Yes, Only use one rwlock may be enough , but if the object lock is removed currently, there will be inconsistencies when creating catalog concurrently, such as failure of MetadataMgr.addMetadata(). I think we need to adjust the scope of writeLock in `createCatalog`. I record a todo.